frontend/src/routers/DeviceRouter.tsx

```diff
@@ -19,12 +21,32 @@ import { LogPage } from '../pages/LogPage'
 
 export const DeviceRouter: React.FC<{ singlePanel?: boolean }> = ({ singlePanel }) => {
   const { deviceID } = useParams<{ deviceID?: string }>()
-  const { device, targetDevice, targets } = useSelector((state: ApplicationState) => ({
+  const { remoteUI, device, targetDevice, targets, fetching } = useSelector((state: ApplicationState) => ({
+    remoteUI: isRemoteUI(state),
+    fetching: state.devices.fetching,
     device: selectDevice(state, deviceID),
     targetDevice: state.backend.device,
     targets: state.backend.targets,
   }))
 
+  const history = useHistory()
+  const { devices } = useDispatch<Dispatch>()
+  const [loaded, setLoaded] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (deviceID && !device && !fetching) {
+      // check that target device is registered and don't redirect
+      if (loaded && !(remoteUI && targetDevice.uid)) {
+        history.push('/devices')
+      } else if (!loaded) {
+        devices.fetchSingle({ id: deviceID, hidden: true })
```

Comment:
do you need await?

Reply:
Having the await caused the graphql load to happen twice. Had it there originally - this works better.